### PR TITLE
[ci] add fluxfile support

### PR DESCRIPTION
**Pull request by vitorvezani**
## Description
add support for helm:
- `fluxFile` (HelmReleases)
- `version`

## Checklist

* [X] I have updated the CHANGELOG of any plugins updated
* [X] I have updated the version file of any plugins updated
* [X] I have added tests wherever appropriate
* [X] I have labeled this PR with a label per plugin updated

### CODE REVIEW

File: plugins/ci/pkg/ci/ci.go [hunk 1]
```diff
@@ -535,13 +534,97 @@ func ProcessHelmTemplates(helmConfigs []models.HelmConfig, tempFolder string, co
...
+	}
+
+	repoDownloadPath := fmt.Sprintf("%s/downloaded-charts/%s/", tempFolder, repoName)
+	chartDownloadPath := repoDownloadPath + chartName
+
+	chartFullName := fmt.Sprintf("%s/%s", repoName, chartName)
+	params := []string{"fetch", chartFullName, "--untar", "--destination", repoDownloadPath}
+
+	version := helmRelease.Spec.Chart.Spec.Version
+	if version != "" {
```
[the remark this answers is not in the document]
Comment (rbren) [reply plugins/ci/pkg/ci/ci.go:607]: Not neccesary I think

File: plugins/ci/pkg/ci/ci.go [hunk 2]
```diff
@@ -535,17 +537,100 @@ func ProcessHelmTemplates(helmConfigs []models.HelmConfig, tempFolder string, co
...
+	err = util.RunCommand(exec.Command("helm", "repo", "add", repoName, helm.Repo), "Adding chart repository: "+repoName)
+	if err != nil {
+		return err
+	}
+
+	repoDownloadPath := fmt.Sprintf("%s/downloaded-charts/%s/", tempFolder, repoName)
+	chartDownloadPath := repoDownloadPath + chartName
+
+	chartFullName := fmt.Sprintf("%s/%s", repoName, chartName)
+	params := []string{"fetch", chartFullName, "--untar", "--destination", repoDownloadPath}
```
Comment (rbren) [inline plugins/ci/pkg/ci/ci.go:604]: Lots of this logic is duplicated. Can we extract the details we need (repo, chart name, version, etc) and pass those to a helper function?

File: plugins/ci/pkg/ci/ci.go [hunk 3]
```diff
@@ -535,17 +537,100 @@ func ProcessHelmTemplates(helmConfigs []models.HelmConfig, tempFolder string, co
 			if err != nil {
 				return err
 			}
+		} else if helm.IsFluxFile() {
```
Comment (rbren) [inline plugins/ci/pkg/ci/ci.go:540]: there's something off about the indentation here 🤔 maybe it's just GitHub

File: plugins/ci/pkg/models/models.go [hunk 4]
```diff
@@ -46,18 +46,24 @@ type HelmConfig struct {
...
 	Chart      string                 `yaml:"chart"`
+	FluxFile   string                 `yaml:"fluxFile"`
+	Version    string                 `yaml:"version"`
 	ValuesFile string                 `yaml:"valuesFile"`
 	Values     map[string]interface{} `yaml:"values"`
 }
 
 func (hc *HelmConfig) IsRemote() bool {
-	return hc.Repo != ""
+	return hc.Repo != "" && !hc.IsFluxFile()
```
Comment (rbren) [inline plugins/ci/pkg/models/models.go:56]: IIUC all the FluxFiles point to a remote repo. For correctness, I think this should be `hc.Repo != "" || hc.IsFluxFile()`

You'd then have to change your `if/else` logic below a bit

File: plugins/ci/pkg/ci/ci.go (hunk 3 above)
Comment (vitorvezani) [reply plugins/ci/pkg/ci/ci.go:540]: Looks like GitHub is to blame on this one

Screen Shot 2021-10-29 at 17 24 48

File: plugins/ci/pkg/models/models.go (hunk 4 above)
Comment (vitorvezani) [reply plugins/ci/pkg/models/models.go:56]: Yes, I think this is not the right abstraction... will try to make it better

File: plugins/ci/pkg/ci/ci.go [hunk 5]
```diff
@@ -74,7 +74,11 @@ func GetAllResources(configDir string, configurationObject models.Configuration)
 				if helm.IsLocal() {
 					displayFilename = filepath.Join(helm.Path, displayFilename)
 				} else if helm.IsRemote() {
-					displayFilename = filepath.Join(helm.Chart, displayFilename)
+					if helm.IsFluxFile() {
+						displayFilename = filepath.Join(helm.Name, displayFilename)
```
Comment (rbren) [inline plugins/ci/pkg/ci/ci.go:78]: in this case, I think displayFilename can just be `helm.FluxFile`

**Review by rbren — APPROVED** [review]

**Pull request merged**